restcomm/restcomm.sms/src/main/java/org/restcomm/connect/sms/smpp/SmppMessageHandler.java

```diff
@@ -364,18 +368,28 @@ public void outbound(SmppOutboundMessageEntity request) throws SmppInvalidArgume
             submit0.setDataCoding(SmppConstants.DATA_CODING_DEFAULT);
             textBytes = CharsetUtil.encode(request.getSmppContent(), SmppClientOpsThread.getOutboundDefaultEncoding());
         }
+        if(autodetectdcs) {
+            submit0.setDataCoding((byte) DATA_CODING_AUTODETECT);
+        }
 
+        boolean payloadFlag = SmppClientOpsThread.getMessagePayloadFlag();
+        int contentLength = request.getSmppContent().length();
         //TODO reverted from https://telestax.atlassian.net/browse/RESTCOMM-1595 as it caused SMS loop at SMSC
         //TODO the delivery receipt should be introduced only together with the remaining/pending DLR implementation
         //TODO the DLR implementation should be configurable (on/off)
         //TODO when enabling delivery receipts again, enable also SmppTest.testClientSentOutUsingSMPPDeliveryReceipt()
         //set the delivery flag to true
         //submit0.setRegisteredDelivery((byte) 1);
 
-        submit0.setShortMessage(textBytes);
-
         TlvSet tlvSet = request.getTlvSet();
 
+        logger.info("payloadFlag="+payloadFlag+" contentLength="+contentLength+" textBytes="+Arrays.toString(textBytes));
+        if(payloadFlag || (contentLength> CONTENT_LENGTH_MAX)) {
```

Comment:
ummm, im not sure this is worthy. Do we want to configure when to use messagePayload TLV instead of regular message body?

I think we should just use messagePayload when content is greater than a paritcular length...(this would be my preffered solution)

In case we really need to make configurable (maybe to interoperate with smpp entities with no support with this TLV), maybe the condition should be &&, so we only use messagePayload when flag/feature is activated and content length is big.

Another approach is to make it configurable and use messagePayload regardless content length (is ther any incovenience if smpp entity supports this TLV).

Reply:
I think the new config is necessary. 

I only see it as two big branches:
-if CONFIG true, always use message_payload
-If CONFIG is false, send message with body 
         -If sending with body and the message length is longer than normal message limit, 
             We have two options:
             -Try sending with other means `submit_multi` or `message_payload`
                    -Since we have not opted in with sending `submit_multi` the only option here is `message_payload`
             -Or Dont send message and flag error.
                    -I think the old implementation mainly tried to stuff everything into the body and mightve caused an exception(?) or the `submit_sm` just didnt get sent. So i assume weve never really opted in into rejecting a message just because the message is too big

So by my assumption where we're not using `submit_multi` nor reject large messages is why i decided `(CONFIG || tooLarge)` instead of (CONFIG && tooLarge)`

Reply:
ok, seems fair explanation,lets proceed